BankClient/Bank.cs

```diff
@@ -1,166 +1,115 @@
-﻿namespace BankClient;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 
+namespace BankClient;
 public class Bank
 {
-    private void menu()
+    public Bank()
     {
-        Console.WriteLine("-------Menu-------");
-        Console.WriteLine("Enter 1 to add client\nEnter 2 to delete client\nEnter 3 to show list of users\nEnter 4 to search one user\nEnter 5 to search group of user\nEnter 6 to change user information");
-        Console.WriteLine("----------gigMenu ends----------");
-        Console.Write("Choose action: ");
+        Users.Add(new UserInformation("Dmitry", "Lapshin", "1234"));
+        Users.Add(new UserInformation("Dima", "Lapshin", "1234"));
     }
 
-    private List<UserInformation> SearchUsersInformation()
-    {
-        Console.WriteLine("Example:Dmitry,Lapshin,0950095521");
-        Console.Write("Fiter: ");
-        var userfilter = Console.ReadLine().Split(',');
-        foreach (var param in userfilter)
-        {
-            param.Trim();
-        }
-
-        if (userfilter[0] == "" && userfilter.Length == 1)
-        {
-            ShowAllUsers();
-        }
-        var filterlist = Users.Where(x => x.FirstName == userfilter[0] || x.LastName == userfilter[0] || x.Phonenumber== userfilter[0] || x.ID.ToString() == userfilter[0]).ToList();
-        for (int i = 0; i < userfilter.Length; i++)
-        {
-            filterlist = filterlist.Where(x => x.FirstName == userfilter[i] || x.LastName == userfilter[i] || x.Phonenumber == userfilter[i] ||  x.ID.ToString() == userfilter[0]).ToList();
-        }
-
-        return filterlist;
-    }
+    private List<UserInformation> Users = new List<UserInformation>();
 
-    public void GetAllUsersInformation(List<UserInformation> filterlist)
+    /// <summary>
+    /// Function saves filterList and call function for print
+    /// </summary>
+    /// <param name="inputUser">Takes <see cref="UserInformation"/> object for filter</param>
+    /// <returns>groupOfUsers</returns>
+    public List<UserInformation> GetAllUsersInformation(UserInformation inputUser)
     {
-        if (filterlist.Count == 0)
+        var filterList = Users.Where(listUsers =>
+            (listUsers.FirstName == inputUser.FirstName || string.IsNullOrEmpty(inputUser.FirstName)) &&
+            (listUsers.LastName == inputUser.LastName || string.IsNullOrEmpty(inputUser.LastName)) &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || string.IsNullOrEmpty(inputUser.PhoneNumber))).ToList();
+        if (filterList.Count == 0)
```

Comment:
Let's remove this check and return empty `filterList` here. It will be empty list in worst case which is ok